crates/driver/src/domain/eth/gas.rs

```diff
@@ -0,0 +1,141 @@
+use {
+    super::{Ether, U256},
+    std::ops,
+};
+
+/// Gas amount.
+#[derive(Debug, Default, Clone, Copy)]
+pub struct Gas(pub U256);
+
+impl From<U256> for Gas {
+    fn from(value: U256) -> Self {
+        Self(value)
+    }
+}
+
+impl From<u64> for Gas {
+    fn from(value: u64) -> Self {
+        Self(value.into())
+    }
+}
+
+impl From<Gas> for U256 {
+    fn from(value: Gas) -> Self {
+        value.0
+    }
+}
+
+/// An EIP-1559 gas price estimate.
+///
+/// https://eips.ethereum.org/EIPS/eip-1559#specification
+#[derive(Debug, Clone, Copy)]
+pub struct GasPrice {
+    /// The maximum total fee that should be charged.
+    pub max: MaxFeePerGas,
+    /// The maximum priority fee (i.e. the tip to the block proposer) that
+    /// can be charged.
+    pub tip: MaxPriorityFeePerGas,
+    /// The current base gas price that will be charged to all accounts on the
+    /// next block.
+    pub base: BaseFeePerGas,
```

Comment:
Not so sure about all these types. `MaxFeePerGas` and `BaseFeePerGas` don't have to be 2 different types as they are the same unit. And I think the same goes for `MaxPriorityFeePerGas` as well.
I would say in this context a type usually represents a unit and the field name + doc comments explains what the value + unit is supposed to represent. Especially since these 3 things only make sense as part of a `GasPrice` and don't get used on their own.
But since we don't really use these types they also don't hurt.

Reply:
The reason I kept them as separate types is that they model slightly different things and have different behaviors in the context of EIP-1559.

I will just unify them to `FeePerGas`, as I agree they may be slightly overkill.